File: instrumentation/http4s-ember-client-2.13_0.23/src/main/scala/com/nr/instrumentation/http4s/InboundResponseWrapper.scala
package com.nr.instrumentation.http4s

import com.newrelic.api.agent.{ExtendedInboundHeaders, HeaderType}
import org.http4s.Headers
import org.typelevel.ci.CIString

import scala.jdk.CollectionConverters._
import java.util

class InboundResponseWrapper[F[_]](headers: Headers) extends ExtendedInboundHeaders {
  /**
    * Return the type of header key syntax used for this.
    *
    * @return An <code>enum</code> specifying the type of headers present.
    * @since 3.5.0
    */
  override def getHeaderType: HeaderType = HeaderType.HTTP

  /**
    * Returns the value of the specified request header as a <code>String</code>. If the request does not include a
    * header with the specified input name, then this method returns <code>null</code>.
    *
    * @param name The name of the desired request header.
    * @return A <code>String</code> containing the value of the specified input request header, or <code>null</code> if the request header is not present.
    * @since 3.5.0
    */
  override def getHeader(name: String): String =
    headers.headers.find(_.name == CIString(name)).map(_.value).orNull

  override def getHeaders(name: String): util.List[String] =
    headers.headers.map(_.name.toString).asJava
}

File: instrumentation/netty-reactor-http-1.0.0/src/main/java/com/nr/instrumentation/InboundResponseWrapper.java
/*
 *
 *  * Copyright 2026 New Relic Corporation. All rights reserved.
 *  * SPDX-License-Identifier: Apache-2.0
 *
 */

package com.nr.instrumentation;

import com.newrelic.api.agent.ExtendedInboundHeaders;
import com.newrelic.api.agent.HeaderType;
import reactor.netty.http.client.HttpClientResponse;

import java.util.List;

public class InboundResponseWrapper extends ExtendedInboundHeaders {

    private final HttpClientResponse response;

    public InboundResponseWrapper(HttpClientResponse response) {
        this.response = response;
    }

    @Override
    public HeaderType getHeaderType() {
        return HeaderType.HTTP;
    }

    @Override
    public String getHeader(String name) {
        List<String> headers = response.responseHeaders().getAll(name);
        if (headers != null && !headers.isEmpty()) {
            return headers.get(0);
        }
        return null;
    }
}